Kernel/Storage/SD/Commands.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace Kernel::SD {
+
+// Commands, defined in PLSS 4.7.4 with the format described in BCM2835 "CMDTM Register"
+enum class CommandIndex : u8 {
+    GoIdleState = 0,
+    AllSendCid = 2,
+    SendRelativeAddr = 3,
+    AppSetBusWidth = 6,
+    SelectCard = 7,
+    SendIfCond = 8,
+    SendCsd = 9,
+    SetBlockLen = 16,
+    ReadSingleBlock = 17,
+    ReadMultipleBlock = 18,
+    WriteSingleBlock = 24,
+    WriteMultipleBlock = 25,
+    AppSendOpCond = 41,
+    AppSendScr = 51,
+    AppCmd = 55,
+};
+
+enum class CommandType : u8 { Normal,
+    Suspend,
+    Resume,
+    Abort };
+
+enum class ResponseType : u8 {
+    NoResponse,
+    ResponseOf136Bits,
+    ResponseOf48Bits,
+    ResponseOf48BitsWithBusy
+};
+
+enum class DataTransferDirection : u8 { HostToCard,
+    CardToHost };
+
+enum class SendAutoCommand : u8 { Disabled,
+    Command12,
+    Command23 };
+
+union Command {
+    u32 raw;
+
+    struct [[gnu::packed]] {
+        u8 : 1;
+        bool block_counter : 1;
+        SendAutoCommand auto_command : 2;
+        DataTransferDirection direction : 1;
+        bool multiblock : 1;
+        u16 : 10;
+        ResponseType response_type : 2;
+        u8 : 1;
+        bool crc_enable : 1;
+        bool idx_enable : 1;
+        bool is_data : 1;
+        CommandType type : 2;
+        CommandIndex index : 6;
+        u8 : 2;
+    };
```

Comment:
prefer to use one type for the bit field type, when its not an enum
like `u32` in the case where the final size is 32 bits

Reply:
Also I think this does not need to be explicitely packed